xds/src/main/java/io/grpc/xds/EdsLoadBalancer.java

```diff
@@ -91,29 +92,40 @@
       LocalityStoreFactory localityStoreFactory,
       Bootstrapper bootstrapper,
       XdsChannelFactory channelFactory) {
-    this.edsLbHelper = edsLbHelper;
-    this.channelLogger = edsLbHelper.getChannelLogger();
-    this.resourceUpdateCallback = resourceUpdateCallback;
-    this.lbRegistry = lbRegistry;
-    this.localityStoreFactory = localityStoreFactory;
+    this.edsLbHelper = checkNotNull(edsLbHelper, "edsLbHelper");
+    this.resourceUpdateCallback = checkNotNull(resourceUpdateCallback, "resourceUpdateCallback");
+    this.lbRegistry = checkNotNull(lbRegistry, "lbRegistry");
+    this.localityStoreFactory = checkNotNull(localityStoreFactory, "localityStoreFactory");
+    this.bootstrapper = checkNotNull(bootstrapper, "bootstrapper");
+    this.channelFactory = checkNotNull(channelFactory, "channelFactory");
     this.switchingLoadBalancer = new GracefulSwitchLoadBalancer(edsLbHelper);
-    this.bootstrapper = bootstrapper;
-    this.channelFactory = channelFactory;
+    logId = InternalLogId.allocate("eds-lb", edsLbHelper.getAuthority());
+    logger = XdsLogger.withLogId(logId);
+    logger.log(XdsLogLevel.INFO, "Created");
   }
 
   @Override
   public void handleResolvedAddresses(ResolvedAddresses resolvedAddresses) {
-    channelLogger.log(ChannelLogLevel.DEBUG, "Received ResolvedAddresses {0}", resolvedAddresses);
-
+    logger.log(XdsLogLevel.DEBUG, "Received resolution result: {0}", resolvedAddresses);
     Object lbConfig = resolvedAddresses.getLoadBalancingPolicyConfig();
-    if (!(lbConfig instanceof XdsConfig)) {
+    if (lbConfig == null) {
       edsLbHelper.updateBalancingState(
           TRANSIENT_FAILURE,
-          new ErrorPicker(Status.UNAVAILABLE.withDescription(
-              "Load balancing config '" + lbConfig + "' is not an XdsConfig")));
+          new ErrorPicker(Status.UNAVAILABLE.withDescription("Missing EDS lb config")));
       return;
     }
     XdsConfig newXdsConfig = (XdsConfig) lbConfig;
+    if (logger.isLoggable(XdsLogLevel.INFO)) {
+      logger.log(
+          XdsLogLevel.INFO,
+          "Received EDS lb config: cluster={0}, child_policy={1}, fallback_policy={2}, "
```

Comment:
`"Received EDS lb config:\n{0}", newXdsConfig`

Reply:
I intended to have INFO level logs being _informative_, descriptive and natural, and DEBUG level logs revealing the real implementation details. Printing out object's toString() always have the square bracket inclosing it. This isn't that comfortable for reading.

Reply:
You'll have to reimplement this message every time the config is change. Look at the refactoring xds design, how complicated the child_policy will be.

Reply:
I don't see much benefit to separately log them in INFO level and DEBUG level, which is quite duplicate.